docs: improve compose example (#10518)
zitadel/zitadel: docs/docs/self-hosting/deploy/compose.mdx

import Next from './_next.mdx'
import NoteInstanceNotFound from './troubleshooting/_note_instance_not_found.mdx';

> This guide is for demonstration and development purposes only.
> It does not set up a production-ready and security hardened instance of Zitadel.
> However, it is your entrypoint for running the Zitadel platform locally for the first time.
> Once you have Zitadel up and running, continue your path to production in the [What's Next section](#next-steps).

Reviewer: Lets use an introduction that start by explaining with what the guide _is_, instead of _not is_. Also, make it less personal be omitting the use of "you".

```suggestion
This guide is the entrypoint for running the Zitadel platform locally for the first time. 
It is for demonstration and development purposes only and does not set up a production-ready and security hardened instance of Zitadel.
Once Zitadel up is and running, learn more about a production setup in the [What's Next section](#next-steps).
```

The setup is likely to work with other software versions too, but it is tested against the following environment:

- Ubuntu 24.04.1
- Docker 28.3.2
- Docker Compose v2.38.2
- curl 8.5.0

## Docker compose

By executing the commands below, you will download the following file:

1. Download the `docker-compose.yaml` file from the Zitadel repository:
   ```shell
   curl -L https://raw.githubusercontent.com/zitadel/zitadel/main/docs/docs/self-hosting/deploy/docker-compose.yaml -o docker-compose.yaml
   ```
2. Make sure you have the latest image versions.
    ```shell
    docker compose pull
    ```
3. Run the PostgreSQL database, the Zitadel API, and the Zitadel login.
    ```shell
    docker compose up --detach --wait
    ```
4. Verify your containers are running and healthy.
    ```shell
    docker compose ps
    ```

<DefaultUser/>

## What's next{#next-steps}

Before proceeding, review the `docker-compose.yaml` file you just downloaded:
The comments give context to the used variables and show examples for commonly used configuration variants.

<details>
    <summary>docker-compose.yaml</summary>
    <CodeBlock language="yaml">{DockerComposeSource}</CodeBlock>
</details>

> Here are some natural steps forward to go from your current setup to production:

Reviewer: ```suggestion
Here are some natural steps forward to go from the current setup to production:
```
